xbuild/src/gradle/mod.rs

```diff
@@ -67,6 +67,36 @@ pub fn build(env: &BuildEnv, libraries: Vec<(Target, PathBuf)>, out: &Path) -> R
         dependencies.push_str(&format!("implementation '{}'\n", dep));
     }
 
+    let mut dexes = String::new();
+    for dex in &env.config().android().dexes {
+        let mut path = env.cargo().package_root().join(dex);
+
+        // Pop the filename and use the directory.
+        //
+        // This is needed as we must provide a directory to `DexMergingTask::dexDirs`
+        path.pop();
```

Comment:
>  Quite unfortunate: how do we control that there are no accidental/stale dex files in the directories?

Right now, I don't. If there is a way to provide only files we should change to that but I haven't found it yet. If that doesn't exist, perhaps we should change the xbuild API to point to a directory in the manifest yaml instead of individual dex files.

> How did you deal with the "gradle does not know who produced this file, and cannot build a dependency graph" sort-of warning/error?

By using a completely different task from yesterday. `DexMergingTask` not `DexArchiveBuilderTask`